pkg/utils/utils.go

```diff
@@ -1175,3 +1175,25 @@ func getUpgradeInfo[T csmv1.CSMComponentType](ctx context.Context, operatorConfi
 	// Example return value: "v2.2.0"
 	return upgradePath.MinUpgradePath, nil
 }
+
+func getNamespaces(ctx context.Context, ctrlClient crclient.Client) ([]string, error) {
+	// Set to store unique namespaces
+	namespaceMap := make(map[string]struct{})
+
+	csmList := &csmv1.ContainerStorageModuleList{}
+
+	if err := ctrlClient.List(ctx, csmList); err != nil {
+		return nil, fmt.Errorf("error listing csm resources: %w", err)
+	}
+	for _, csmResource := range csmList.Items {
+		namespaceMap[csmResource.Namespace] = struct{}{}
+	}
+
+	// Convert set to slice
```

Comment:
why are we converting the map to a slice? why not just use the slice from the beginning and append to it?

Reply:
It's to make sure only unique namespace entries are added. Using a map will ensure that. Duplicates will be removed